TodoList/src/app/screens/BookList/components/Book/index.js

```diff
@@ -1,29 +1,49 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { View, Text, Image } from 'react-native';
+import {
+  View, Text, Image, TouchableOpacity
+} from 'react-native';
 import styles from './styles';
+import books from '../../../../../constants/books';
+import routes from '../../../../../constants/routes';
 
-function Book({ image, title, author }) {
+function Book({
+  image, title, author, id, navigation
+}) {
+  const navigateToDetails = () => {
```

Comment:
If you have an inner function, you must turn this component into a class component.
Also, since the component has logic now, turn it into smart & dumb components.